polaris/hub/polarisfs.py

```diff
@@ -120,3 +129,73 @@ def cat_file(
         with fsspec.open(signed_url, "rb", **kwargs) as f:
             data = f.read()
         return data[start:end]
+
+    def rm(self, path: str, recursive: bool = False, maxdepth: Optional[int] = None) -> None:
+        """Remove a file or directory from the Polaris dataset.
+
+        This method is provided for compatibility with the Zarr storage interface.
+        It may be called by the Zarr store when removing a file or directory.
+
+        Args:
+            path: The path to the file or directory to be removed.
+            recursive: If True, remove directories and their contents recursively.
+            maxdepth: The maximum depth to recurse when removing directories.
+
+        Returns:
+            None
+
+        Note:
+            This method currently it does not perform any removal operations and is included
+            as a placeholder that aligns with the Zarr interface's expectations.
+        """
+        raise NotImplementedError("PolarisFS does not currently support the file removal operation.")
+
+    def pipe_file(
+        self,
+        path: str,
+        content: Union[bytes, str],
+        timeout: Optional[TimeoutTypes] = None,
+        **kwargs: dict,
+    ) -> None:
+        """Pipes the content of a file to the Polaris dataset.
+
+        Args:
+            path: The path to the file within the dataset.
+            content: The content to be piped into the file.
+            timeout: Maximum time (in seconds) to wait for the request to complete.
+
+        Returns:
+            None
+        """
+        if timeout is None:
+            timeout = self.default_timeout
+
+        pipe_path = self.sep.join([self.base_path, path])
+
+        # PUT request to Polaris Hub to put object in path
+        response = self.polaris_client.put(pipe_path, timeout=timeout, content=content)
+
+        if response.status_code != 307:
+            raise PolarisHubError("Could not get signed URL from Polaris Hub.")
+
+        hub_response_body = response.json()
+        signed_url = hub_response_body["url"]
+
+        sha256_hash = hashlib.sha256(content).hexdigest()
+
+        headers = {
+            "Content-Type": "application/octet-stream",
+            "x-amz-content-sha256": sha256_hash,
+            "x-amz-date": datetime.datetime.utcnow().strftime("%Y%m%dT%H%M%SZ"),
```

Comment:
`utcnow()` is [being deprecated](https://blog.miguelgrinberg.com/post/it-s-time-for-a-change-datetime-utcnow-is-now-deprecated), so this is causing warnings! We should instead use:

```python
from datetime import datetime, timezone
datetime.now(timezone.utc)
```